eventually/authentication/models.py

```diff
@@ -102,32 +134,43 @@ def myattr_to_dict(self):
             'updated_at': int(self.updated_at.timestamp())
             }
 
-    #'FIXME: rewrite docs'
-    def update(self,
-               first_name=False,
-               last_name=False,
-               middle_name=False,
-               email=False,
-               password=False):
+    def update(self, first_name=None, last_name=None, middle_name=None, email=None, password=None):
         """
         Updates user profile in the database with the specified parameters.\n
         :param first_name: <str> User's first name
         :param last_name: <str> User's last name
         :param middle_name: <str> User's middle
         :param email: <str> User's  email
         :param password: <str> User's password
-        :return: returns True if object updated or False if parameters were incorrect
+        :return: returns True if object updated, False if parameters were incorrect
+                 or -1 if none of the conditions succeded.
         """
-        user = CustomUser.get_by_id(self.id)
+        user_changed = False
+        if self.id:
+            user = CustomUser.get_by_id(self.id)
+        else:
+            print("You're trying to modify a user which does not exist in the database")
+            return False
         if first_name:
             user.first_name = first_name
+            user_changed = True
         if last_name:
-            user.first_name = last_name
+            user.last_name = last_name
+            user_changed = True
         if middle_name:
             user.middle_name = middle_name
+            user_changed = True
         if email:
             user.email = email
+            user_changed = True
         if password:
             user.password = password
-        user.save()
-        return True
+            user_changed = True
+        if user_changed:
+            try:
+                user.save()
+                return True
+            except django.db.utils.IntegrityError:
+                print('ERR: A user with such an email already exists.')
```

Comment:
`ERR:` what does it mean? probably error?

Reply:
Yes, it's short for error. Nevertheless, all prints were removed.